src/dao/TIMEHolderDAO.js

```diff
@@ -19,8 +28,20 @@ export default class TIMEHolderDAO extends AbstractContractDAO {
   async deposit (amount: number) {
     const assetDAO = await this.getAssetDAO()
     const account = await this.getAddress()
-    await assetDAO.approve(account, amount)
-    return this._tx(TX_DEPOSIT, [assetDAO.addDecimals(amount)], {amount})
+
+    // dryRuns
+    const [gas1, gas2] = await Promise.all([
+      await assetDAO.pluralApprove(account, amount, {isDryRun: true}),
+      await this.pluralDeposit(assetDAO, amount, {isDryRun: true})
+    ])
+
+    // confirm and run tx
+    await assetDAO.pluralApprove(account, amount, {step: 1, of: 2, gasLeft: gas1 + gas2})
+    return this.pluralDeposit(assetDAO, amount, {step: 2, of: 2, gasLeft: gas2})
```

Comment:
don't start second tx suddenly, show to user pinned notification and start it only when he will click on this notification